api/envoy/config/route/v3/route_components.proto

```diff
@@ -317,6 +317,21 @@ message Route {
   // If set, the bytes actually buffered will be the minimum value of this and the
   // listener per_connection_buffer_limit_bytes.
   google.protobuf.UInt32Value per_request_buffer_limit_bytes = 16;
+
+  // The human readable prefix to use when emitting statistics for this endpoint.
+  // The statistics are rooted at vhost.<virtual host name>.route.<stat_prefix>.
+  // This should be set for highly critical
+  // endpoints that one wishes to get “perfect” statistics on.
+  // If not set, endpoint statistics are not generated.
+  //
+  // The emitted statistics are the same as those documented for :ref:`virtual clusters <config_http_filters_router_vcluster_stats>`.
+  //
+  // .. warning::
+  //
+  //    We do not recommend setting up a stat prefix for
+  //    every application endpoint. This is both not easily maintainable and
+  //    statistics use a non-trivial amount of memory.
```

Comment:
You can give a rough cost estimate here -- I'd estimate 100 bytes per counter/gauge, which you then multiply by the number of statistics per route. Looks like about 8 stats per route, so 800 bytes per route? Is there another multiplier?

Reply:
Yes. per route it is 800Bytes. There is no other multiplier here.